src/olympia/files/tests/test_models.py

```diff
@@ -1075,6 +1075,22 @@ def test_webextension(self):
                                  parse_data={'is_webextension': True})
         assert file_.is_webextension
 
+    def test_webextension_zip(self):
+        """Test to ensure we accept ZIP uploads, but convert them into XPI
+        files ASAP to keep things simple.
+        """
+        upload = self.upload('webextension.zip')
+        file_ = File.from_upload(upload, self.version, self.platform,
+                                 parse_data={'is_webextension': True})
+        assert file_.filename.endswith('.xpi')
+        assert file_.is_webextension
+
+    def test_extension_zip(self):
+        upload = self.upload('recurse.zip')
+        file_ = File.from_upload(upload, self.version, self.platform)
+        assert file_.filename.endswith('.xpi')
+        assert not file_.is_webextension
```

Comment:
I'm not sure if it's meant for this but you could use `self.check_delete(file_, file_.file_path)` to clean up the files.